quickstart/backend/src/main/java/com/digitalasset/quickstart/service/LicenseApiImpl.java

```diff
@@ -55,135 +65,197 @@ public LicenseApiImpl(
+                                                            l.setContractId(contract.contractId.getContractId);
+                                                            l.setDso(contract.payload.getDso.getParty);
+                                                            l.setProvider(contract.payload.getProvider.getParty);
+                                                            l.setUser(contract.payload.getUser.getParty);
+
+                                                            org.openapitools.model.LicenseParams lp = new org.openapitools.model.LicenseParams();
+                                                            org.openapitools.model.Metadata meta = new org.openapitools.model.Metadata();
+                                                            meta.setData(contract.payload.getParams.getMeta.getValues);
+                                                            lp.setMeta(meta);
+                                                            l.setParams(lp);
+
+                                                            l.setExpiresAt(OffsetDateTime.ofInstant(contract.payload.getExpiresAt, ZoneOffset.UTC));
+                                                            l.setLicenseNum(contract.payload.getLicenseNum.intValue());
+                                                            return l;
+                                                        })
+                                                        .collect(Collectors.toList());
+                                                return ResponseEntity.ok(result);
+                                            });
+                                })
+                        ).thenCompose(cf -> cf)
+                )
+                .whenComplete(
+                        completeWithin(parentContext, (res, ex) -> {
+                            if (ex == null) {
+                                int count = (res.getBody() != null) ? res.getBody().size() : 0;
+                                Map<String, Object> successAttributes = Map.of("count", count);
+                                LoggingSpanHelper.logDebug(logger, "listLicenses: success", successAttributes);
+                            } else {
+                                LoggingSpanHelper.logError(logger, "listLicenses: failed", startAttributes, ex);
+                                LoggingSpanHelper.recordException(methodSpan, ex);
+                            }
+                        })
+                );
     }
 
     @Override
+    @WithSpan
     public CompletableFuture<ResponseEntity<Void>> renewLicense(
-            String contractId,
-            String commandId,
-            org.openapitools.model.LicenseRenewRequest licenseRenewRequest
+            @SpanAttribute("license.contractId") String contractId,
+            @SpanAttribute("license.commandId") String commandId,
```

Comment:
These are `commandId` and `contractId` everywhere else in the logging, so having them be distinct here is a problem when querying them.

Also see expireLicense above.